stable/fluentd/templates/deployment.yaml

```diff
@@ -12,17 +12,20 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
+{{- if .Values.autoscaling.enabled}}
+  serviceName: {{ template "fluentd.fullname" . }}
+{{- end}}
 {{- if not .Values.autoscaling.enabled }}
   replicas: {{ .Values.replicaCount }}
 {{- end }}
   selector:
     matchLabels:
       app: {{ template "fluentd.name" . }}
       release: {{ .Release.Name }}
-  {{- if .Values.persistence.enabled }}
-  strategy:
-    type: Recreate
-  {{- end }}
+{{- if .Values.persistence.enabled }}
```

Comment:
This probably won't work. If you run a deployment with one replica and a PVC with `ReadWriteOnce`, you have to go with `Recreate`. Otherwise, a new pod will never come up because the PVC is still attached to the old pod.

Reply:
Done. Please check.